### PR TITLE
CASMHMS-5461 - Adds HSM re-discovery after creds update

### PR DESCRIPTION
**Pull request by mbuchmann-hpe**
### Summary and Scope

After credential changes on BMCs, the credentials on nodes are not updated.  Capmc (and maybe other services) uses node credentials.  Re-discovery of the BMC updates the credentials for the nodes.  This change triggers a re-discovery of the BMC after a credential change.

DOES THIS CHANGE INVOLVE ANY SCHEME CHANGES?  N

REMINDER: HAVE YOU INCREMENTED VERSION NUMBERS? 1.10.0

REMINDER 2: HAVE YOU UPDATED THE COPYRIGHT PER hpe GUIDELINES: (C) Copyright 2014-2021 Hewlett Packard Enterprise Development LP    ? Y

### Issues and Related PRs

CASMHMS-5461

### Testing

LIST THE ENVIRONMENTS IN WHICH THESE CHANGES WERE TESTED.

Tested on:

* WASP

Ran tests on several nodes, verified that the HSM discovery process was started.  Tested some error conditions.

### Risks and Mitigations

Low risk - 

Requires:

### CODE REVIEW

File: cmd/scsd/creds.go [hunk 1]
```diff
@@ -1112,3 +1130,49 @@ func doCredsGet(w http.ResponseWriter, r *http.Request) {
...
+		hsmTransport = &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}
+		hsmClient = &http.Client{Transport: hsmTransport}
+	}
+	url := appParams.SmdURL + "/Inventory/Discover"
+	logger.Error(url)
+
+	var payloadS discoverPayload
+	payloadS.Xnames = xnames
+	payloadS.Force = true
+	//	payload := "{\"xnames\":[\"" + xname + "\"], \"force\":true}"
```
Comment (rsjostrand-hpe) [inline cmd/scsd/creds.go:1148]: Can we remove this comment, I don't think we need it?

File: cmd/scsd/creds.go [hunk 2]
```diff
@@ -1112,3 +1130,49 @@ func doCredsGet(w http.ResponseWriter, r *http.Request) {
 	w.Write(ba)
 }
 
+func doHSMDiscover(xnames []string) ([]byte, error) {
+	if len(xnames) == 0 {
+		return nil, nil
+	}
+	logger.Error("********** IN HSM DISCOVER ****************")
```
Comment (rsjostrand-hpe) [inline cmd/scsd/creds.go:1137]: Maybe we can change this log message to be more informational? Something like:
```
logger.Println("Informing HSM to discover the following redfish endpoints:, strings.Join(xnames, ",")
```

**Review by rsjostrand-hpe — CHANGES_REQUESTED** [review]

File: cmd/scsd/creds.go (hunk 1 above)
Comment (mbuchmann-hpe) [reply cmd/scsd/creds.go]: Updated

File: cmd/scsd/creds.go (hunk 2 above)
Comment (mbuchmann-hpe) [reply cmd/scsd/creds.go]: Removed

**Review by rsjostrand-hpe — APPROVED** [review]

**Review by shunr-hpe — APPROVED** [review]

**Pull request merged**